LayoutTests/fast/canvas/webgl/webgl2-invalid-passed-params.html

```diff
@@ -0,0 +1,46 @@
+<html>
+<head>
+<script src="../../../resources/js-test.js"></script>
+<script src="resources/webgl-test.js"></script>
+<script src="resources/desktop-gl-constants.js"></script>
+</head>
+<body>
+<div id="description"></div>
+<div id="console"></div>
+
+<script>
+    description("Test for webgl2 invalid passed parameters");
+
+    if (window.internals)
+        window.internals.settings.setWebGLErrorsToConsoleEnabled(false);
+
+    var gl = create3DContext(null, null, true);
+
+    debug("Invalid gl.texImage3D() usage");
+    var texture = gl.createTexture();
+    gl.bindTexture (gl.TEXTURE_3D, texture);
+
+    debug("gl.INVALID_VALUE is generated if width, height or depth is greater than gl.MAX_3D_TEXTURE_SIZE.");
+    const tooBig3DTextureSize = gl.getParameter(gl.MAX_3D_TEXTURE_SIZE) + 1;
+    shouldGenerateGLError(gl, gl.INVALID_VALUE, "gl.texImage3D(gl.TEXTURE_3D, 0, gl.RGBA, tooBig3DTextureSize, 1, 1, 0, gl.RGBA, gl.UNSIGNED_BYTE, null)");
+    shouldGenerateGLError(gl, gl.INVALID_VALUE, "gl.texImage3D(gl.TEXTURE_3D, 0, gl.RGBA, 1, tooBig3DTextureSize, 1, 0, gl.RGBA, gl.UNSIGNED_BYTE, null)");
+    shouldGenerateGLError(gl, gl.INVALID_VALUE, "gl.texImage3D(gl.TEXTURE_3D, 0, gl.RGBA, 1, 1, tooBig3DTextureSize, 0, gl.RGBA, gl.UNSIGNED_BYTE, null)");
+    shouldGenerateGLError(gl, gl.INVALID_VALUE, "gl.texImage3D(gl.TEXTURE_3D, 0, gl.RGBA, tooBig3DTextureSize, tooBig3DTextureSize, tooBig3DTextureSize, 0, gl.RGBA, gl.UNSIGNED_BYTE, null)");
+    gl.deleteTexture(texture);
+
+    texture = gl.createTexture();
+    gl.bindTexture (gl.TEXTURE_2D_ARRAY, texture);
+
+    debug("");
+    debug("gl.INVALID_VALUE is generated if width, height or depth is greater than gl.MAX_TEXTURE_SIZE.");
+    const tooBigTextureSize = gl.getParameter(gl.MAX_TEXTURE_SIZE) + 1;
+    shouldGenerateGLError(gl, gl.INVALID_VALUE, "gl.texImage3D(gl.TEXTURE_2D_ARRAY, 0, gl.RGBA, tooBigTextureSize, 1, 1, 0, gl.RGBA, gl.UNSIGNED_BYTE, null)");
+    shouldGenerateGLError(gl, gl.INVALID_VALUE, "gl.texImage3D(gl.TEXTURE_2D_ARRAY, 0, gl.RGBA, 1, tooBigTextureSize, 1, 0, gl.RGBA, gl.UNSIGNED_BYTE, null)");
+    shouldGenerateGLError(gl, gl.INVALID_VALUE, "gl.texImage3D(gl.TEXTURE_2D_ARRAY, 0, gl.RGBA, 1, 1, tooBigTextureSize, 0, gl.RGBA, gl.UNSIGNED_BYTE, null)");
+    shouldGenerateGLError(gl, gl.INVALID_VALUE, "gl.texImage3D(gl.TEXTURE_2D_ARRAY, 0, gl.RGBA, tooBigTextureSize, tooBigTextureSize, tooBigTextureSize, 0, gl.RGBA, gl.UNSIGNED_BYTE, null)");
+    gl.deleteTexture(texture);
```

Comment:
Please consider a more focused test here.

This snippet is a copy of an existing dEQP WebGL test that accidentally triggers the WebKit issue because `MAX_3D_TEXTURE_SIZE` is usually 4096, which leads to an overflow on line 28. All other `texImage3D` calls here do not bring any value.

It would be great to test all four affected entry points (`tex[Sub]Image{2|3}D`) instead with dimensions that are guaranteed to cause integer overflow, not depending on `getParameter`.